torch/distributions/distribution.py

```diff
@@ -28,9 +29,12 @@ def __init__(self, batch_shape=torch.Size(), event_shape=torch.Size(), validate_
             self._validate_args = validate_args
         if self._validate_args:
             for param, constraint in self.arg_constraints.items():
-                if not constraints.is_dependent(constraint):
-                    if not constraint.check(self.__getattribute__(param)).all():
-                        raise ValueError("The parameter {} has invalid values".format(param))
+                if constraints.is_dependent(constraint):
+                    continue  # skip constraints that cannot be checked
+                if param not in self.__dict__ and isinstance(getattr(type(self), param, None), lazy_property):
```

Comment:
+1! Thanks for checking this. I have thought that there is no way to check lazy_property without trigger it. Good to know about `getattr(type(obj),...)`.